src/communication/bolt/v1/encoder/base_encoder.hpp

```diff
@@ -264,6 +295,7 @@ class BaseEncoder {
 
  protected:
   Buffer &buffer_;
+  int major_v_;  //!< Major version of the underlying Bolt protocol (TODO: Think about reimplementing the versioning)
```

Comment:
Maybe specify what in your view should be changed and how if you have some idea (briefly)

Reply:
Added a comment. Basically if we ever refactor the higher-level and end up with classes per bolt version, we could template this and have each bolt version use the correct version out of the box.